bugsnag-android-core/src/main/java/com/bugsnag/android/Configuration.java

```diff
@@ -154,86 +153,69 @@ public void setContext(@Nullable String context) {
     }
 
     /**
-     * Get the endpoint to send data
-     *
-     * @return Endpoint
+     * @deprecated use {@link Configuration#getEndpoints()}
      */
+    @Deprecated
     @NonNull
     public String getEndpoint() {
-        return endpoint;
+        return endpoints.getNotify();
     }
 
     /**
-     * Set the endpoint to send data to. By default we'll send reports to
-     * the standard https://notify.bugsnag.com endpoint, but you can override
-     * this if you are using Bugsnag Enterprise to point to your own Bugsnag
-     * endpoint.
-     *
-     * @param endpoint the custom endpoint to send report to
-     * @deprecated use {@link com.bugsnag.android.Configuration#setEndpoints(String, String)}
+     * @deprecated use {@link Configuration#setEndpoints(Endpoints)}
      */
     @Deprecated
     public void setEndpoint(@NonNull String endpoint) {
-        this.endpoint = endpoint;
+        setEndpoints(new Endpoints(endpoint, getEndpoints().getSessions()));
+    }
+
+    /**
+     * @deprecated use {@link Configuration#setEndpoints(Endpoints)}
+     */
+    @Deprecated
+    public void setEndpoints(@NonNull String notify, @NonNull String sessions) {
+        setEndpoints(new Endpoints(notify, sessions));
     }
 
     /**
      * Set the endpoints to send data to. By default we'll send error reports to
      * https://notify.bugsnag.com, and sessions to https://sessions.bugsnag.com, but you can
-     * override this if you are using Bugsnag Enterprise to point to your own Bugsnag endpoint.
-     *
-     * Please note that it is recommended that you set both endpoints. If the notify endpoint is
-     * missing, an exception will be thrown. If the session endpoint is missing, a warning will be
-     * logged and sessions will not be sent automatically.
-     *
-     * @param notify the notify endpoint
-     * @param sessions the sessions endpoint
+     * override this if you are using Bugsnag Enterprise to point to your own Bugsnag endpoints.
      *
-     * @throws IllegalArgumentException if the notify endpoint is empty or null
+     * @param endpoints the notify and sessions endpoint
      */
-    public void setEndpoints(@NonNull String notify, @NonNull String sessions)
-        throws IllegalArgumentException {
-
-        if (TextUtils.isEmpty(notify)) {
-            throw new IllegalArgumentException("Notify endpoint cannot be empty or null.");
-        } else {
-            this.endpoint = notify;
-        }
-
-        boolean invalidSessionsEndpoint = TextUtils.isEmpty(sessions);
-
-        if (invalidSessionsEndpoint) {
-            Logger.warn("The session tracking endpoint has not been set. "
-                + "Session tracking is disabled");
-            this.sessionEndpoint = null;
-            this.autoCaptureSessions = false;
-        } else {
-            this.sessionEndpoint = sessions;
-        }
+    public void setEndpoints(@NonNull Endpoints endpoints) {
```

Comment:
why is this better than `setEndpoints(@NonNull String notify, @NonNull String sessions)`?

Reply:
I think I'm going to answer my own question. It more closely matches the data structure you'd load from the config file.